test/benchmarks/CMakeLists.txt

```diff
@@ -41,3 +42,43 @@ if(ZLIB_ENABLE_TESTS)
     add_test(NAME benchmark_zlib
         COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:benchmark_zlib>)
 endif()
+
+if (WITH_BENCHMARK_APPS)
+    option(BUILD_ALT_BENCH "Link against alternative zlib implementation" OFF)
+
+    # Search for libpng package
+    find_package(PNG QUIET REQUIRED)
+
+    set(BENCH_APP_SRCS 
+        benchmark_png_encode.cc
+        benchmark_png_decode.cc
+        benchmark_main.cc
+    )
+
+    add_executable(benchmark_zlib_apps ${BENCH_APP_SRCS})
+
+    if (DEFINED BUILD_ALT_BENCH)
+        set(ZLIB_ALT_LIB "libz.a" CACHE FILEPATH "Optional alternative zlib implementation (defaults to stock zlib)")
```

Comment:
I almost wonder if we should use `FetchContent` for madler/zlib too. This would mean that it would not build the alternative app if `ZLIB_COMPAT=ON`, but that would be totally fine imo. If zlib-compat is OFF, then it would only build the one app...

Reply:
But I wonder if that would work or not. Because the project names are so similar.

Reply:
Marking as resolved for now.  The suggestion about a parallel fetch for zlib is likely to cause problems given that they are exported "zlib" as the project name in both places.  We only actually need the library itself and I generally expect the user to specify a path to this if not the distro installed zlib on a Linux distribution.